aspnetcore/release-notes/aspnetcore-5.0.md

```diff
@@ -0,0 +1,464 @@
+
+### Component virtualization
+
+Improve the perceived performance of component rendering using the Blazor framework's built-in virtualization support. For more information, see <xref:blazor/forms-validation#radio-buttons>.
+
+### `ontoggle` event support
+
+Blazor events now support the `ontoggle` DOM event. For more information, see <xref:blazor/components/event-handling#event-argument-types>.
+
+### Set UI focus in Blazor apps
+
+Use the `FocusAsync` convenience method on element references to set the UI focus to that element. For more information, see <xref:blazor/components/event-handling#focus-an-element>.
+
+### Custom validation class attributes
+
+Custom validation class names are useful when integrating with CSS frameworks, such as Bootstrap. For more information, see <xref:blazor/forms-validation#custom-validation-class-attributes>.
+
+### IAsyncDisposable support
+
+Blazor components now support the <xref:System.IAsyncDisposable> interface for the asynchronous release of allocated resources.
+
+### JavaScript isolation and object references
+
+Blazor enables JavaScript isolation in standard [JavaScript modules](https://developer.mozilla.org/docs/Web/JavaScript/Guide/Modules). For more information, see <xref:blazor/call-javascript-from-dotnet#blazor-javascript-isolation-and-object-references>.
+
+### Form components support display name
+
+The following built-in components support display names with the `DisplayName` parameter:
+
+* `InputDate`
+* `InputNumber`
+* `InputSelect`
+
+For more information, see <xref:blazor/forms-validation#display-name-support>.
+
+### Catch-all route parameters
+
+Catch-all route parameters, which capture paths across multiple folder boundaries, are supported in components. For more information, see <xref:blazor/fundamentals/routing#catch-all-route-parameters>.
+
+### Debugging improvements
+
+Debugging Blazor WebAssembly apps is improved in ASP.NET Core 5.0. Additionally, debugging is now supported on Visual Studio for Mac. For more information, see <xref:blazor/debug>.
+
+### Microsoft Identity v2.0 and MSAL v2.0
+
+Blazor security now uses Microsoft Identity v2.0 ([`Microsoft.Identity.Web`](https://www.nuget.org/packages/Microsoft.Identity.Web) and [`Microsoft.Identity.Web.UI`](https://www.nuget.org/packages/Microsoft.Identity.Web.UI)) and MSAL v2.0. For more information, see the topics in the [Blazor Security and Identity node](xref:blazor/security/index).
+
+### State management support
+
+`Microsoft.AspNetCore.ProtectedBrowserStorage` has been moved to `Microsoft.AspNetCore.Components.Web.Extensions` and has undergone API changes. State management documentation has been split across Blazor hosting models. For more information, see <xref:blazor/state-management>.
+
+<!-- WAIT ON THIS ENTRY UNTIL https://github.com/dotnet/AspNetCore.Docs/pull/19887 MERGES.
+
+### Component integration across hosting models
```

Comment:
This same functionality also enabled prerendering of Blazor WebAssembly apps. I think Blazor WebAssembly prerendering is signficant enough that it should have its own section.